### PR TITLE
#4017: Editing for Kafka ResetOffset example

### LINKED ISSUE #4017: Docs writer update for Kafka example PR
Style updates and editing for https://github.com/knative/docs/pull/4012

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #4017

### CODE REVIEW

File: config/nav.yml [hunk 1]
```diff
@@ -214,6 +214,7 @@ nav:
           - Overview: eventing/samples/kafka/README.md
           - Binding Example: eventing/samples/kafka/binding/README.md
           - Channel Example: eventing/samples/kafka/channel/README.md
+          - ResetOffset Example: eventing/samples/kafka/resetoffset/README.md
```
Comment (abrennan89) [inline config/nav.yml:217]: @travis-minke-sap added to nav

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 2]
```diff
@@ -1,91 +1,83 @@
...
+Kafka-backed Channels differ from other Channel implementations because they provide temporal persistence of events.
+
+Event persistence enables certain advanced use cases to be supported, such as the ability to replay prior events by repositioning their current location in the event stream. This feature is useful when attempting to recover from unexpected Subscriber downtime.
+
+The [ResetOffset custom resource definition (CRD)](https://github.com/knative-sandbox/eventing-kafka/tree/main/config/command/resetoffset) exposes the ability to manipulate the location of the ConsumerGroup Offsets in the event stream of a given Knative Subscription. Without the ResetOffset CRD, you must manually stop ConsumerGroups and manipulating Offsets.
 
 !!! note
     ResetOffsets are currently only supported by the Distributed KafkaChannel
-    implementation. It is expected that the Consolidated KafkaChannel
-    implementation will support them in the near future.
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/resetoffset/README.md:30]: @travis-minke-sap we don't really talk about what might happen in the future in docs, just FYI 🙂

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 3]
```diff
@@ -1,91 +1,83 @@
...
+- A Kubernetes cluster with the [Kafka Channel implementation](https://knative.dev/docs/eventing/channels/channels-crds/) installed.
+- A valid KafkaChannel resource and Subscription exist.
 
-The following examples assume a Kubernetes Namespace called `my-namespace`
-containing a KafkaChannel Subscription named `my-subscription` and should be
-updated according to your use case.
+!!! note
+    The ResetOffset CRD is a single-use operation, and should be
+    deleted from the cluster once it has been executed.
+    <!--TODO: Explain why?-->
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/resetoffset/README.md:26]: @travis-minke-sap do you know why this is? I haven't heard of any other single use CRDs in Knative

File: docs/eventing/samples/kafka/resetoffset/README.md (hunk 2 above)
Comment (travis-minke-sap) [reply docs/eventing/samples/kafka/resetoffset/README.md:30]: 👍

File: config/nav.yml (hunk 1 above)
Comment (travis-minke-sap) [reply config/nav.yml:217]: Thank you!

File: docs/eventing/samples/kafka/resetoffset/README.md (hunk 3 above)
Comment (travis-minke-sap) [reply docs/eventing/samples/kafka/resetoffset/README.md:26]: Yes, this resource is a bit different.  The feature went through pretty extensive investigation / design / review in order to achieve agreement that this was the "best" way forward.  Essentially we are specifying a one-time operation (e.g. move the offsets once right now and then be done) in the standard Kubernetes declarative fashion.  The README linked at the top of this examples page provides more information.

Maybe something like the following would be better?

```
The ResetOffset CRD is a single-use operation that should be monitored and removed once it has successfully completed.  Previously successful ResetOffset instances will not be re-executed, but are clutter that must be ignored.
```

Comment (abrennan89) [reply docs/eventing/samples/kafka/resetoffset/README.md:26]: I think it's actually less confusing as is 🙂 I was just curious as to why

**Comment by travis-minke-sap** [comment]
Pretty sure I don't have the requisite permissions to merge this but... fwiw ; )
/lgtm

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 4]
```diff
@@ -1,91 +1,82 @@
...
 
-!!! note
-    Repositioning the ConsumerGroup Offsets will impact the event ordering and
-    is intended for failure recovery scenarios. This capability needs to be used
-    with caution only after reviewing the CRD documentation linked above.
+Kafka-backed Channels differ from other Channel implementations because they provide temporal persistence of events.
+
+Event persistence enables certain advanced use cases to be supported, such as the ability to replay prior events by repositioning their current location in the event stream. This feature is useful when attempting to recover from unexpected Subscriber downtime.
+
+The [ResetOffset custom resource definition (CRD)](https://github.com/knative-sandbox/eventing-kafka/tree/main/config/command/resetoffset) exposes the ability to manipulate the location of the ConsumerGroup Offsets in the event stream of a given Knative Subscription. Without the ResetOffset CRD, you must manually stop ConsumerGroups and manipulating Offsets.
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:7]:
```suggestion
The [ResetOffset custom resource definition (CRD)](https://github.com/knative-sandbox/eventing-kafka/tree/main/config/command/resetoffset) exposes the ability to manipulate the location of the ConsumerGroup Offsets in the event stream of a given Knative Subscription. Without the ResetOffset CRD, you must manually stop ConsumerGroups and manipulate Offsets.
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 5]
```diff
@@ -1,91 +1,82 @@
...
 
-## Examples
+- A Kubernetes cluster with the [Kafka Channel implementation](https://knative.dev/docs/eventing/channels/channels-crds/) installed.
+- A valid KafkaChannel resource and Subscription exist.
 
-The following examples assume a Kubernetes Namespace called `my-namespace`
-containing a KafkaChannel Subscription named `my-subscription` and should be
-updated according to your use case.
+!!! note
+    The ResetOffset CRD is a single-use operation, and should be
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:24]:
```suggestion
    The ResetOffset CRD is a single-use operation, and must be
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 6]
```diff
@@ -1,91 +1,82 @@
...
+  namespace: <namespace>
 spec:
   offset:
     time: earliest
   ref:
     apiVersion: messaging.knative.dev/v1
     kind: Subscription
-    namespace: my-namespace
-    name: my-subscription
+    namespace: <subcription-namespace>
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:44]:
```suggestion
    namespace: <subscription-namespace>
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 7]
```diff
@@ -1,91 +1,82 @@
...
+    name: <subscription-name>
 ```
 
-### Repositioning the Offsets to a specific point in the event stream
+Where:
+
+- `<resetoffset-name>` is the name of the ResetOffset CRD.
+- `<namespace>` is the namespace where you want to create the ResetOffset CRD.
+- `<subcription-namespace>` is the namespace where your Subscription exists.
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:52]:
```suggestion
- `<subscription-namespace>` is the namespace where your Subscription exists.
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 8]
```diff
@@ -1,91 +1,82 @@
...
 
 !!! note
     ResetOffsets are currently only supported by the Distributed KafkaChannel
-    implementation. It is expected that the Consolidated KafkaChannel
-    implementation will support them in the near future.
+    implementation.
 
-## Prerequisites
+!!! warning
+    Repositioning the ConsumerGroup Offsets will impact the event ordering and
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:14]:
```suggestion
    Repositioning the ConsumerGroup Offsets impacts the event ordering and
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 9]
```diff
@@ -1,91 +1,82 @@
...
 spec:
   offset:
-    time: "2021-06-17T17:30:00Z"
+    time: <offset-time>
   ref:
     apiVersion: messaging.knative.dev/v1
     kind: Subscription
-    namespace: my-namespace
-    name: my-subscription
+    namespace: <subcription-namespace>
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:72]:
```suggestion
    namespace: <subscription-namespace>
```

File: docs/eventing/samples/kafka/resetoffset/README.md [hunk 10]
```diff
@@ -1,91 +1,82 @@
...
 
-!!! note
-    The ResetOffset CRD is a single use operation or "command", and should be
-    deleted from the cluster once it has completed execution.
+Where:
+
+- `<resetoffset-name>` is the name of the ResetOffset CRD.
+- `<offset-time>` is the specified offset time. You will need to set an [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339) timestamp relative to your use case, for example `"2021-06-17T17:30:00Z"`.
+- `<namespace>` is the namespace where you want to create the ResetOffset CRD.
+- `<subcription-namespace>` is the namespace where your Subscription exists.
```
Comment (snneji) [inline docs/eventing/samples/kafka/resetoffset/README.md:81]:
```suggestion
- `<subscription-namespace>` is the namespace where your Subscription exists.
```

**Review by snneji — CHANGES_REQUESTED** [review]
Fixed a few minor typos. Other than that looks great.

File: docs/eventing/samples/kafka/resetoffset/README.md (hunk 5 above)
Comment (abrennan89) [reply docs/eventing/samples/kafka/resetoffset/README.md:24]: I actually asked the same thing, but this is a recommendation, not mandatory AFAIK, so I'll leave it as "should" for the moment, even though I know we'd prefer to not do that...

**Comment by travis-minke-sap** [comment]
great catches on the typos/grammer - thanks again!

**Review by snneji — APPROVED** [review]

**Pull request merged**